### PR TITLE
Check for Vulnerabilities via GHE Action

### PR DESCRIPTION
**Pull request by lukasniemeier-zalando**
Follow-up of https://github.com/zalando/logbook/pull/1742

See https://github.com/zalando/logbook/actions/runs/7613871782 for example.

### CODE REVIEW

**Comment by lukasniemeier-zalando** [comment]
:+1:

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -29,6 +29,8 @@ jobs:
     - name: Dependency Review
       uses: actions/dependency-review-action@v4
       with:
+        base-ref: ${{ github.event.pull_request.base.sha || '0.1.0' }}
```
Comment (kasmarian) [inline .github/workflows/build.yaml:32]: In https://github.com/actions/dependency-review-action it says that `base-ref/head-ref` is used for event types other than `pull_request` and `pull_request_target`. Why do we need the `github.event.pull_request.base.sha` and `github.event.pull_request.head.sha` then?

Comment (lukasniemeier-zalando) [reply .github/workflows/build.yaml:32]: as far as I can see their documentation is imprecise, see https://github.com/actions/dependency-review-action/blob/935098a9508277d66b6c3adc56de47383d4c1e56/src/git-refs.ts#L10-L11

Comment (kasmarian) [reply .github/workflows/build.yaml:32]: Interesting, so if we define `base-ref` and `head-ref` they'll override PRs events as well.

**Comment by lukasniemeier-zalando** [comment]
The (bit ugly) trick of running non-pull-request builds by comparing the current state with the early tag seems to work - https://github.com/zalando/logbook/actions/runs/7623503585/job/20763667755 looked for vulnerabilities across all dependencies :heavy_check_mark:

**Pull request merged**